runtime/gc_modron_startup/mmparseXXgc.cpp

```diff
@@ -655,6 +657,7 @@ gcParseXXgcArguments(J9JavaVM *vm, char *optArg)
 				returnValue = JNI_EINVAL;
 				break;
 			}
+			extensions->configuration->_cacheListSplitForced = true;
```

Comment:
I missed this detail, when introduced in OMR, but Forced flags have always been in Extensions, not In Configuration. But then, we put too many things in Extensions, too, so in a long run I might be ok with having them in Configuration (and we should even consider moving the other Forced ones, and even beyond Forced ones).
I still don't like is having _cacheListSplitForced in base Configuration, while it's specific to Config Generational (additionally, whole class is under SCAVENGE ifdef, while this field is not). It would required upcasting to fetch it, but it's still reasonable.

Reply:
I agree that it's specific to config generational, I think the reason why I introduced it in base class was because the base class was already dealing with the actual init of this gen specific value:

```C++
MM_Configuration::initializeGCParameters
#if defined(OMR_GC_MODRON_SCAVENGER)
	/* initialize scan cache lock splitting factor */
	if (!_cacheListSplitForced) {
		extensions->cacheListSplit = OMR_MAX(extensions->cacheListSplit, splitAmount);
	}
	if (extensions->scavengerEnabled) {
		if (MM_GCExtensionsBase::OMR_GC_SCAVENGER_SCANORDERING_NONE == extensions->scavengerScanOrdering) {
			extensions->scavengerScanOrdering = MM_GCExtensionsBase::OMR_GC_SCAVENGER_SCANORDERING_HIERARCHICAL;
		} else if (MM_GCExtensionsBase::OMR_GC_SCAVENGER_SCANORDERING_DYNAMIC_BREADTH_FIRST == extensions->scavengerScanOrdering) {
			extensions->adaptiveGcCountBetweenHotFieldSort = true;
		}
	}
#endif /* OMR_GC_MODRON_SCAVENGER */
```

Hence, this forced flag would need to be visible at the base class layer. Looking back, it would of been wiser to extract cacheSplit specifics out and shift to config gen

Reply:
I would prefer to get these changes in and then do a set of refactoring changes right after

Reply:
To minimize the amount of commits/changes to be double delivered for 0.40